dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -2092,4 +2148,38 @@ public Response handleGitHubBranchDeletion(@ApiParam(hidden = true) @Parameter(h
         githubWebhookDelete(repository, gitReference, username);
         return Response.status(HttpStatus.SC_NO_CONTENT).build();
     }
+
+    @GET
+    @Path("/{workflowId}/workflowVersions/{workflowVersionId}/orcidAuthors")
+    @UnitOfWork(readOnly = true)
+    @ApiOperation(value = "Retrieve ORCID author information for a workflow version",  hidden = true)
+    @Operation(operationId = "getWorkflowVersionOrcidAuthors", description = "Retrieve ORCID author information for a workflow version", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @ApiResponse(responseCode = HttpStatus.SC_OK + "", description = "Retrieve ORCID author information for a workflow version", content = @Content(
+            mediaType = MediaType.APPLICATION_JSON, array = @ArraySchema(schema = @Schema(implementation = OrcidAuthorInformation.class))))
+    @ApiResponse(responseCode = HttpStatus.SC_BAD_REQUEST + "", description = "Bad Request")
+    public Set<OrcidAuthorInformation> getWorkflowVersionOrcidAuthors(@Parameter(hidden = true, name = "user")@Auth User user,
+            @Parameter(name = "workflowId", description = "id of the workflow", required = true, in = ParameterIn.PATH)  @PathParam("workflowId") Long workflowId,
+            @Parameter(name = "workflowVersionId", description = "id of the workflow version", required = true, in = ParameterIn.PATH) @PathParam("workflowVersionId") Long workflowVersionId) {
+        Workflow workflow = workflowDAO.findById(workflowId);
+        checkEntry(workflow);
+        checkCanRead(user, workflow);
+
+        WorkflowVersion workflowVersion = this.workflowVersionDAO.findById(workflowVersionId);
+        if (workflowVersion == null) {
+            throw new CustomWebApplicationException("Version " + workflowVersionId + " does not exist for this workflow", HttpStatus.SC_NOT_FOUND);
+        }
+
+        Set<OrcidAuthorInformation> orcidAuthorInfo = new HashSet<>();
+        Optional<String> token = ORCIDHelper.getOrcidAccessToken();
+        if (token.isPresent()) {
+            orcidAuthorInfo = workflowVersion.getOrcidAuthors().stream()
```

Comment:
As a followup issue, you could have an in-memory map of fetched authors. One with a fairly short expiration, like 10 minutes or so. Then if a user is flipping between versions, subsequent calls will probably be pretty snappy (I wouldn't think the ORCID ids would change too much between versions), and it would reduce the number of calls we make to ORCID.

Reply:
Did some quick testing and you're right the subsequent calls are much faster. For a version with 4 orcid authors, subsequent calls are 30 - 40 ms vs the 1.5 - 2 s for the initial request. This is also useful because it seems like orcid authors are fetched every time a workflow version is published/unpublished, snapshotted, DOI'd, and when a default version is set.

Will create a follow up ticket